src/test/rbd_mirror/test_ImageDeleter.cc

```diff
@@ -0,0 +1,210 @@
+// -*- mode:C; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2016 SUSE LINUX GmbH
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+#include "include/rados/librados.hpp"
+#include "include/rbd/librbd.hpp"
+#include "include/stringify.h"
+#include "cls/rbd/cls_rbd_client.h"
+#include "tools/rbd_mirror/ImageDeleter.h"
+#include "tools/rbd_mirror/ImageReplayer.h"
+#include "tools/rbd_mirror/Threads.h"
+#include "librbd/ImageCtx.h"
+#include "librbd/internal.h"
+#include "librbd/Utils.h"
+
+#include "test/librados/test.h"
+#include "gtest/gtest.h"
+
+#include <thread>
+#include <chrono>
+
+#define GLOBAL_IMAGE_ID "global_image_id"
+
+#define dout_subsys ceph_subsys_rbd_mirror
+
+using rbd::mirror::RadosRef;
+
+void register_test_rbd_mirror_image_deleter() {
+}
+
+class TestImageDeleter : public ::testing::Test {
```

Comment:
If you inherit from the "TestFixture" class, it creates and connects to the two pools automatically.